### PR TITLE
0.15 Release Notes: Retained Rendering

### PR DESCRIPTION
**Pull request by Trashtalk217**
This truly is a terrible first draft, but I had fun writing it and I will probably improve it in the future.

### CODE REVIEW

**Review by cart — APPROVED** [review]
I think you hit all of the right notes here, but there are some issues with this first draft:

1. Too wordy: I think we can express the same things in about half the space
2. "Too much personality" / told from your perspective: I think your wording and tone is fun, but I also think it comes at the cost of clarity. A few too many "in jokes" and implied context. And in one place you mention "yours truly". In general, the release notes are written by "us" not "me"
3. "Too much history":  theres too much coverage of the details of how this was implemented (how PRs played out, bugs fixed after the initial prs, etc).

I've reworded it to address these issues. Feel free to tweak further, provided it doesn't regress on that front.

**Comment by Trashtalk217** [comment]
@cart I never claimed it was any good. Thank you for fixing it up. I've since added the migration guide. It's probably a bit long (the second 'Extract components with `ExtractComponentPlugin`' is technically superfluous), but I think it covers everything.

Also, I'd like to credit the authors of all the original prs: @kristoff3r, @re0132 and @tychedelia and I'm not quite sure where that goes.

**Comment by alice-i-cecile** [comment]
> Also, I'd like to credit the authors of all the original prs: @kristoff3r, @re0312 and @tychedelia and I'm not quite sure where that goes.

This goes in the `release-notes.md` at the top of that folder :) There's a section for this metadata for each release note.

File: release-content/0.15/migration-guides/15320_Retained_Rendering.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+) {
+    for (render_entity, clusters, camera) in &views {
+        // some code
+        // After the sync step, all main world entities with a &RenderEntity have a corresponding (empty) render world entity. This should never panic.
+        commands.entity(render_entity).insert(...);
+    }
+}
+
+// in main world, when spawning
+world.spawn(Clusters::default(), Camera::default(), SyncToRenderWorld)
```
Comment (JMS55) [inline release-content/0.15/migration-guides/15320_Retained_Rendering.md:57]:
```suggestion
world.spawn((Clusters::default(), Camera::default(), SyncToRenderWorld))
```

File: release-content/0.15/migration-guides/15320_Retained_Rendering.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+) {
+    for (main_entity, clusters, camera in &views) {
+        // do something
+    }
+}
+```
+
+### General advice for working with main and render world entities
+
+When working with entities from both worlds it can be confusing. If you are every in a scenario where this isn't entirely clear (for example, when working on custom extraction code in the render world), we advice that you use `RenderEntity` and `MainEntity` as simple wrappers around `Entity`. Mixing these up can become a real headache and lead to some non-obvious errors.
```
Comment (JMS55) [inline release-content/0.15/migration-guides/15320_Retained_Rendering.md:77]:
```suggestion
When working with entities from both worlds it can be confusing. If you are every in a scenario where this isn't entirely clear (for example, when working on custom extraction code in the render world), we advise that you use `RenderEntity` and `MainEntity` as simple wrappers around `Entity`. Mixing these up can become a real headache and lead to some non-obvious errors.
```

**Review by alice-i-cecile — APPROVED** [review]
